src/coreComponents/physicsSolvers/multiphysics/CoupledReservoirAndWellsBase.cpp

```diff
@@ -110,6 +110,47 @@ addCouplingNumNonzeros( SolverBase const * const solver,
   } );
 }
 
+bool validateWellPerforations( SolverBase const * const solver,
+                               WellSolverBase const * const wellSolver,
+                               DomainPartition const & domain )
+{
+  std::pair< string, string > badPerforation;
+  solver->forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const &,
+                                                                        MeshLevel const & meshLevel,
+                                                                        arrayView1d< string const > const & regionNames )
+  {
+    ElementRegionManager const & elemManager = meshLevel.getElemManager();
+    elemManager.forElementSubRegions< WellElementSubRegion >( regionNames, [&]( localIndex const, WellElementSubRegion const & subRegion )
```

Comment:
I have a suggestion to slightly change the way you loop over wells. In the current way, you loop over all the target regions of the coupled solver, then filter them out to get the well region. 

Instead, you could just loop over the well regions by doing: 
```suggestion
  wellSolver->forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const &,
                                                                        MeshLevel const & meshLevel,
                                                                        arrayView1d< string const > const & regionNames )
  {
    ElementRegionManager const & elemManager = meshLevel.getElemManager();
    elemManager.forElementSubRegions< WellElementSubRegion >( regionNames, [&]( localIndex const, WellElementSubRegion const & subRegion )
```